generator/internal/parser/httprule/lexer.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package httprule
+
+import (
+	"fmt"
+	"strings"
+	"unicode/utf8"
+)
+
```

Comment:
I've split this into a separate file because the logic here is pretty generic, and I didn't want it to distract from the business logic in the parser.